### PR TITLE
docs: Add k3s to the Compatibility matrix

### PR DESCRIPTION
**Pull request by onedr0p**
Add k3s nuances to Compatibility matrix

Fixes https://github.com/XenitAB/spegel/issues/212

### CODE REVIEW

File: docs/COMPATIBILITY.md [hunk 1]
```diff
@@ -21,3 +21,4 @@ Spegel has been tested on the following Kubernetes flavors for compatibility. So
 | EKS | :yellow_circle: | Discard unpacked layers is enabled by default and needs to be disabled. |
 | GKE | :red_circle: | Not supported due to registry config path not being set. |
 | Minikube | :green_circle: | Verified to work out of the box. |
+| k3s | :yellow_circle: | Requires setting values in the Helm chart. Set `.spegel.containerdSock` to `/run/k3s/containerd/containerd.sock` and `spegel.containerdRegistryConfigPath` to `/var/lib/rancher/k3s/agent/etc/containerd/certs.d` See [#212](https://github.com/XenitAB/spegel/issues/212)https://github.com/XenitAB/spegel/issues/212 for more information |
```
Comment (onedr0p) [inline docs/COMPATIBILITY.md:24]:
```suggestion
| k3s | :yellow_circle: | Requires setting values in the Helm chart. Set `.spegel.containerdSock` to `/run/k3s/containerd/containerd.sock` and `spegel.containerdRegistryConfigPath` to `/var/lib/rancher/k3s/agent/etc/containerd/certs.d` See [#212](https://github.com/XenitAB/spegel/issues/212) for more information |
```

File: docs/COMPATIBILITY.md [hunk 2]
```diff
@@ -21,3 +21,4 @@ Spegel has been tested on the following Kubernetes flavors for compatibility. So
 | EKS | :yellow_circle: | Discard unpacked layers is enabled by default and needs to be disabled. |
 | GKE | :red_circle: | Not supported due to registry config path not being set. |
 | Minikube | :green_circle: | Verified to work out of the box. |
+| k3s | :yellow_circle: | Requires setting values in the Helm chart. Set `.spegel.containerdSock` to `/run/k3s/containerd/containerd.sock` and `spegel.containerdRegistryConfigPath` to `/var/lib/rancher/k3s/agent/etc/containerd/certs.d` See [#212](https://github.com/XenitAB/spegel/issues/212) for more information |
```
Comment (onedr0p) [inline docs/COMPATIBILITY.md:24]:
```suggestion
| k3s | :yellow_circle: | Requires setting values in the Helm chart. Set `.spegel.containerdSock` to `/run/k3s/containerd/containerd.sock` and `.spegel.containerdRegistryConfigPath` to `/var/lib/rancher/k3s/agent/etc/containerd/certs.d` See [#212](https://github.com/XenitAB/spegel/issues/212) for more information |
```

File: docs/COMPATIBILITY.md [hunk 3]
```diff
@@ -21,3 +21,21 @@ Spegel has been tested on the following Kubernetes flavors for compatibility. So
...
+    ```toml
+    [plugins."io.containerd.grpc.v1.cri".registry]
+      config_path = "/var/lib/rancher/k3s/agent/etc/containerd/certs.d"
+    ```
+
+3. Restart k3s on all nodes
+
+4. In the Spegel Helm chart set `.spegel.containerdSock` to `/run/k3s/containerd/containerd.sock` and `.spegel.containerdRegistryConfigPath` to `/var/lib/rancher/k3s/agent/etc/containerd/certs.d`
+
+5. See [#212](https://github.com/XenitAB/spegel/issues/212) for more information
```
Comment (onedr0p) [inline docs/COMPATIBILITY.md:41]: I am not sure why this is red, but it renders out fine.

**Comment by phillebaba** [comment]
There is currently a PR underway that would make k3s support a lot simpler.

**Comment by onedr0p** [comment]
Oh wow that's fresh off the block, thanks for linking.

**Comment by brandond** [comment]
I have a secret plan to embed spegel in k3s. I don't know if it'll go anywhere (I haven't talked to product management or anyone else on the team), but I'd love to make it work.

**Comment by onedr0p** [comment]
That would be amazing, hopefully you can get approval and time to work on it. I know Talos is working on something similar to have this built into their distro too.

**Comment by phillebaba** [comment]
@brandond we should talk. While I do not have this use case it is one that I have discussed with others. In theory I see no reason why this would not work, but it would be very interesting to see how it would perform in the real world.

**Comment by brandond** [comment]
@phillebaba the biggest hurdle at the moment is the fact that all of the packages are `internal`. 

K3s likes to embed components and run them in its main process - we successfully do this for etcd and all the various Kubernetes components, for example. Unfortunately, I can't take [the bits from I need from main.go](https://github.com/XenitAB/spegel/blob/main/main.go#L125-L169) and embed them directly in k3s because of the internal package restrictions enforced by go - so my first move (at the moment) would involve forking this repo and renaming all the packages.

**Comment by brandond** [comment]
@phillebaba do y'all have a slack channel workspace, or can I find you on Kubernetes slack?

I have some questions about the DHT lookups; I'm seeing that they just don't work at all due to a defect in the async provider lookup channel closure detection logic, but I want to make sure I'm not doing something wrong first.

**Comment by phillebaba** [comment]
@brandond i messages you on the cncf slack. You can also find me on the Kubernetes slack if that's easier.

**Pull request closed without merging**